tests/lib/ocpp/v201/test_charge_point.cpp

```diff
@@ -762,11 +761,34 @@ TEST_F(ChargePointFixture, K01FR29_SmartChargingCtrlrAvailableIsFalse_RespondsCa
     auto set_charging_profile_req =
         request_to_enhanced_message<SetChargingProfileRequest, MessageType::SetChargingProfile>(req);
 
-    EXPECT_CALL(*smart_charging_handler, validate_and_add_profile(testing::_, testing::_)).Times(0);
+    EXPECT_CALL(*smart_charging_handler, validate_and_add_profile).Times(0);
 
     charge_point->handle_message(set_charging_profile_req);
 }
 
+TEST_F(ChargePointFixture, K05FR05_RequestStartTransactionRequest_SmartChargingCtrlrEnabledTrue_ValidatesTxProfiles) {
+    const auto cv = ControllerComponentVariables::SmartChargingCtrlrEnabled;
+    this->device_model->set_value(cv.component, cv.variable.value(), AttributeEnum::Actual, "true", "TEST", true);
+
+    auto periods = create_charging_schedule_periods({0, 1, 2});
+
+    auto profile = create_charging_profile(
+        DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::TxProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), DEFAULT_TX_ID);
+
+    RequestStartTransactionRequest req;
+    req.evseId = DEFAULT_EVSE_ID;
+    req.idToken = IdToken{.idToken = "Local", .type = IdTokenEnum::Local};
+    req.chargingProfile = profile;
+
+    auto start_transaction_req =
+        request_to_enhanced_message<RequestStartTransactionRequest, MessageType::RequestStartTransaction>(req);
+
+    EXPECT_CALL(*smart_charging_handler, validate_and_add_profile).Times(1);
+
+    charge_point->handle_message(start_transaction_req);
+}
```

Comment:
I see this pattern repeated in these tests.

In addition to checking that `validate_and_add_profile` is called, I would suggest simple validation tests verifying the expected output. This could be at least checking that the profile is in the database or similar...

We do have such tests further along - e.g. in `ChargepointTestFixtureV201,` but it doesn't hurt to check again, at a higher level in the system, here.